engine/source/public/core/asset_manager.hpp

```diff
@@ -0,0 +1,76 @@
+﻿#pragma once
+#include "core/asset.hpp"
+
+#include <memory>
+#include <unordered_map>
+
+namespace lumina
+{
+    class Asset;
+    class Engine;
+
+    class AssetManager
+    {
+    private:
+        friend class Engine;
+
+        std::unordered_map<uint32_t, std::shared_ptr<Asset>> assets {};
+
+    public:
+        template <typename T, typename... Args>
+        std::shared_ptr<T> CreateAsset(Args&&... args)
```

Comment:
should args be `Args&&...` and not `Args&...`?

Reply:
Which it is? Or do you mean the other way around?

Reply:
It's currently `&&` and i'm not sure if that is correct or not, because i think it should be `&`. I have never used `&&` before, so i night have missed the reason why you used the double ref (`&&`)

Reply:
&& is a universal reference, and can accept both lvalues or rvalues. Used together with std::forward it preserves the value  category.

Reply:
Because rider is saying `Forwarding reference parameter 'args' is never forwarded inside the function body`. But I don't understand universal reference enough to be sure if this is correct or not.